app/Actions/Profile/UpdateLogin.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace App\Actions\Profile;
+
+use App\Exceptions\ConfigurationKeyMissingException;
+use App\Exceptions\ConflictingPropertyException;
+use App\Exceptions\Internal\QueryBuilderException;
+use App\Models\User;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Log;
+
+class UpdateLogin
+{
+	/**
+	 * Update Username if it does not already exists.
+	 *
+	 * @param User   $user
+	 * @param string $username
+	 * @param string $ip
+	 *
+	 * @return User
+	 *
+	 * @throws ConfigurationKeyMissingException
+	 * @throws QueryBuilderException
+	 * @throws ConflictingPropertyException
+	 */
+	public function updateUsername(User &$user, string $username, string $ip): User
+	{
+		// No need to change
+		if ($username === $user->username) {
+			return $user;
+		}
+
+		// Check if username already exists
+		if (User::query()->where('username', '=', $username)->where('id', '!=', $user->id)->count() !== 0) {
+			Log::channel('login')->warning(__METHOD__ . ':' . __LINE__ . sprintf('User (%s) tried to change their identity to (%s) from %s', $user->username, $username, $ip));
+			throw new ConflictingPropertyException('Username already exists.');
+		}
+
+		// Change username
+		if ($username !== $user->username) {
+			Log::channel('login')->notice(__METHOD__ . ':' . __LINE__ . sprintf('User (%s) changed their identity for (%s) from %s', $user->username, $username, $ip));
```

Comment:
Consistency with the message above
```suggestion
			Log::channel('login')->notice(__METHOD__ . ':' . __LINE__ . sprintf('User (%s) changed their identity to (%s) from %s', $user->username, $username, $ip));
```